dask/dataframe/shuffle.py

```diff
@@ -255,9 +258,12 @@ def shuffle(df, index, shuffle=None, npartitions=None, max_branch=32, compute=No
 
 def rearrange_by_divisions(df, column, divisions, max_branch=None, shuffle=None):
     """ Shuffle dataframe so that column separates along divisions """
+    series_type = type(df[df.columns[0]]._meta)
```

Comment:
Same comment about duplicates. May be worth moving to a helper in `dask.datframe.utils` at this point.

Reply:
Makes sense - Is something like this reasonable?:

```python
def series_like_df(df, values=()):
    """Get a series that is compatible with df.
    """
    return type(df._meta.iloc[:, 0])(values)
```

Reply:
I was thinking just the `series_type` part, not the construction.

```python
def series_type(meta):
    return type(meta.iloc[:, 0])
```

Though, I see now that this isn't well defined for a DataFrame with no columns...

Reply:
Pandas calls this `meta._constructor_sliced`

For a pandas.DataFrame, that returns a pandas.Series.

If pandas is OK with documenting that, then perhaps cudf could do the same and we just use that.

Reply:
Thanks @TomAugspurger!  I think it definitely makes sense to add `_constructor_sliced` to cudf, especially if pandas is willing to document it :)